pkg/controller/util/node/controller_utils.go

```diff
@@ -135,8 +135,9 @@ func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
 		}
 
 		for i, cond := range pod.Status.Conditions {
-			if cond.Type == v1.PodReady {
+			if cond.Type == v1.PodReady && cond.Status != v1.ConditionFalse {
 				pod.Status.Conditions[i].Status = v1.ConditionFalse
+				pod.Status.Conditions[i].LastTransitionTime = metav1.Now()
```

Comment:
Could we replace this with `pkg/api/v1/pod/util.go#UpdatePodCondition`? I think it includes all of the updates we want (i.e. updating `LastTransitionTime`) and has the advantage of centralizing the logic for updating pod condition in a central place.

Reply:
updated!